addons/event_crm/models/event_lead_rule.py

```diff
@@ -186,6 +186,31 @@ def _run_on_registrations(self, registrations):
 
         return self.env['crm.lead'].create(lead_vals_list)
 
+    def action_execute_rule(self):
```

Comment:
This seems a bit sad to duplicate the code / messages for this.

In addition, we do not leverage the "power" of the CRON job that someone worked hard to implement (hum :disguised_face: ).
Meaning you could click on this button and get a timeout...

Wild idea: can't we complement the "action_generate_leads" method on event.event with an optional "event_lead_rules" parameter, that would condition the rules on which it applies?
You would likely have to add the same parameter to "_apply_lead_generation_rules", and add a "event_lead_rule_ids" field on "event.lead.request" as well.

A bit of extra work but probably worth it.

Reply:
Additional bonus: we would not need an extra test (or just tweak the existing one?), meaning it would likely be a smaller diff.